src/track/beats.cpp

```diff
@@ -60,23 +60,23 @@ Beats::ConstIterator Beats::ConstIterator::operator+=(Beats::ConstIterator::diff
     }
 
     DEBUG_ASSERT(n > 0);
-    const int beatOffset = m_beatOffset + n;
 #ifdef MIXXX_DEBUG_ASSERTIONS_ENABLED
     const auto origValue = m_value;
 #endif
 
-    // Detect integer overflow
-    if (beatOffset < m_beatOffset) {
+    // Detect integer overflow in `m_beatOffset + n`
+    const int maxBeatOffset = std::numeric_limits<Beats::ConstIterator::difference_type>::max();
```

Comment:
So this is 2,147,483,647. My track as a ~150 BPM track with 2:21. 
No clue how that can be triggered.

Reply:
I was a bit confused about this too, but apparently the max/min integers are used as a form of marker values to indicate the first/last possible beat (outside of a track):

https://github.com/mixxxdj/mixxx/blob/2c2e706b4419794e1b29f23f124ce65f49b88bc7/src/track/beats.h#L199-L210

Haven't dug too much into the code to find out where that actually is useful, but for now I think coding defensively in the iterator is probably a good idea.